components/post_view/post_list/index.jsx

```diff
@@ -0,0 +1,448 @@
+             */
+            getPostThread: PropTypes.func.isRequired,
+
+            /*
+             * Get unreads posts onload
+             */
+            loadUnreads: PropTypes.func.isRequired,
+
+            /*
+             * Get posts using BEFORE_ID and AFTER_ID
+             */
+            loadPosts: PropTypes.func.isRequired,
+
+            /*
+             * Used for clearing posts from channel
+             */
+            clearPostsFromChannel: PropTypes.func.isRequired,
+
+            /*
+             * Used for changing posts status of channel
+             * For notifying redux store when we hit bottom or top of posts
+             */
+            channelPostsStatus: PropTypes.func.isRequired,
+
+            /*
+             * Used for taking backup of postsInChannel before clearing so,
+             * we can use it incase of failure in loading posts
+             */
+            backUpPostsInChannel: PropTypes.func.isRequired,
+
+            /*
+             * Used for notifying redux store on channel sync complete
+             */
+            channelSyncCompleted: PropTypes.func.isRequired,
+
+            /*
+             * Used for adding back postIds to postsInChannel from backUpPostsInChannel
+             */
+            addPostIdsFromBackUp: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDoingInitialLoad: this.shouldLoadPosts(props),
+            newerPosts: {
+                loading: false,
+                allLoaded: (props.channelPostsStatus && props.channelPostsStatus.atEnd) || false,
+            },
+            olderPosts: {
+                loading: false,
+                allLoaded: this.oldestMessageLoadedInView(props),
+            },
+            lastViewedAt: props.lastViewedAt || props.member.last_viewed_at,
+        };
+    }
+
+    componentDidMount() {
+        if (this.state.isDoingInitialLoad) {
+            this.postsOnLoad(this.props.channel.id);
+        }
+        if (!this.props.channelSyncStatus && this.props.channelPostsStatus) {
+            this.syncPosts(this.props);
+        }
+    }
+
+    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
+        if (this.props.posts && nextProps.posts && (this.props.posts.length !== nextProps.posts.length)) {
+            if (this.oldestMessageLoadedInView(nextProps) !== this.state.olderPosts.allLoaded) {
+                this.setState({
+                    olderPosts: {
+                        loading: false,
+                        allLoaded: this.oldestMessageLoadedInView(nextProps),
+                    },
+                });
+            }
+        }
+        if (!this.props.socketStatus.connected && nextProps.socketStatus.connected) {
+            this.syncPosts(nextProps);
+        }
+    }
+
+    shouldLoadPosts(props) {
+        if (typeof props.channelPostsStatus === 'undefined') {
```

Comment:
I think `if (!props.channelPostsStatus)` will work the same.  Same with the other instance of `undefined`

Reply:
Ya, Initially it was different as it was a flag. False could have evaluated true to this so hence the logic. But this is now an object so i can change this. Good catch thanks!